upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -711,4 +722,999 @@ pub(crate) mod protocol_test {
 
         assert_eq!(ds1_message, ds1_message_second_time.unwrap());
     }
+
+    /// Test that after giving up on a downstairs, setting it to faulted, and
+    /// letting it reconnect, live repair occurs. Check that each extent is
+    /// repaired with the correct source, and that extent limits are honoured if
+    /// additional IO comes through.
+    #[tokio::test]
+    async fn test_successful_live_repair() {
+        let harness = Arc::new(TestHarness::new().await);
+
+        let (jh1, mut ds1_messages) =
+            harness.ds1().await.spawn_message_receiver().await;
```

Comment:
why does the one look different `().await.` than the other two?

Reply:
`ds1` is a function because the harness holds it as an Option to enable it to be `take`n out and disconnected for test purposes. For `ds2` and `ds3`, they're only fields.